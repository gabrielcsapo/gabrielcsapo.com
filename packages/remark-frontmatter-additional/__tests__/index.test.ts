import { describe, test, expect } from "vitest";

import path from "path";
import fs from "node:fs/promises";
import { compile } from "@mdx-js/mdx";
import { VFile } from "vfile";

import remarkFrontmatter from "remark-frontmatter";
import remarkFrontmatterAdditional from "../src/index";

describe("remark-frontmatter-additional", () => {
  test("basic", async () => {
    const content = await fs.readFile(
      path.resolve(__dirname, "fixtures", "post.mdx")
    );
    const file = new VFile(content);
    file.history = ["/src/pages/post.mdx"];

    const { value } = await compile(file, {
      remarkPlugins: [remarkFrontmatter, remarkFrontmatterAdditional],
      jsx: true,
    });

    expect(value).toMatchSnapshot();
  });

  test("file with no title (expecting default first header to exist)", async () => {
    const content = await fs.readFile(
      path.resolve(__dirname, "fixtures", "post-with-no-title.mdx")
    );
    const file = new VFile(content);
    file.history = ["/src/pages/post-with-no-title.mdx"];

    const { value } = await compile(file, {
      remarkPlugins: [remarkFrontmatter, remarkFrontmatterAdditional],
      jsx: true,
    });

    expect(value).toMatchSnapshot();
  });
});
