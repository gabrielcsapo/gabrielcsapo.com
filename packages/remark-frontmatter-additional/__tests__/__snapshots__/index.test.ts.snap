// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`remark-frontmatter-additional > basic 1`] = `
"/*@jsxRuntime automatic @jsxImportSource react*/
export const title = \\"My First Post\\", hello = \\"frontmatter\\", date = \\"2023-03-14\\", image = \\"https://cdn2.thecatapi.com/images/bdc.jpg\\";
export const excerpt = \\"A blockquote spanning multiple lines. \\\\nA second line in the blockquote.\\";
export const slug = \\"/post\\";
export const readingTime = {
  \\"text\\": \\"1 min read\\",
  \\"minutes\\": 0.555,
  \\"time\\": 33300,
  \\"words\\": 111
};
function _createMdxContent(props) {
  const _components = Object.assign({
    h1: \\"h1\\",
    p: \\"p\\",
    em: \\"em\\",
    strong: \\"strong\\",
    code: \\"code\\",
    blockquote: \\"blockquote\\",
    ul: \\"ul\\",
    li: \\"li\\",
    ol: \\"ol\\",
    a: \\"a\\",
    img: \\"img\\",
    pre: \\"pre\\"
  }, props.components);
  return <><_components.h1>{\\"Random Blog Post\\"}</_components.h1>{\\"\\\\n\\"}<_components.p><_components.em>{\\"Italic\\"}</_components.em>{\\", \\"}<_components.strong>{\\"bold\\"}</_components.strong>{\\", and \\"}<_components.code>{\\"monospace\\"}</_components.code>{\\".\\"}</_components.p>{\\"\\\\n\\"}<_components.blockquote>{\\"\\\\n\\"}<_components.p>{\\"A blockquote spanning multiple lines.\\\\nA second line in the blockquote.\\"}</_components.p>{\\"\\\\n\\"}</_components.blockquote>{\\"\\\\n\\"}<_components.ul>{\\"\\\\n\\"}<_components.li>{\\"An unordered list item\\"}</_components.li>{\\"\\\\n\\"}<_components.li>{\\"Another unordered list item\\"}{\\"\\\\n\\"}<_components.ol>{\\"\\\\n\\"}<_components.li>{\\"A nested ordered list item\\"}</_components.li>{\\"\\\\n\\"}<_components.li>{\\"Another nested ordered list item\\"}</_components.li>{\\"\\\\n\\"}</_components.ol>{\\"\\\\n\\"}</_components.li>{\\"\\\\n\\"}</_components.ul>{\\"\\\\n\\"}<_components.p><_components.a href=\\"https://www.example.com\\">{\\"An inline-style link\\"}</_components.a></_components.p>{\\"\\\\n\\"}<_components.p><_components.img src=\\"https://via.placeholder.com/150\\" alt=\\"An image\\" /></_components.p>{\\"\\\\n\\"}<_components.p>{\\"Some inline \\"}<_components.code>{\\"code\\"}</_components.code>{\\".\\"}</_components.p>{\\"\\\\n\\"}<_components.pre><_components.code className=\\"language-js\\">{\\"// Some code in a fenced code block\\\\nfunction hello() {\\\\n  console.log(\\\\\\"Hello, world!\\\\\\");\\\\n}\\\\n\\"}</_components.code></_components.pre>{\\"\\\\n\\"}<_components.p>{\\"| Syntax    | Description |   Test Text |\\\\n| :-------- | :---------: | ----------: |\\\\n| Header    |    Title    | Here's this |\\\\n| Paragraph |    Text     |    And more |\\"}</_components.p>{\\"\\\\n\\"}<_components.p>{\\"~Strikethrough~\\"}</_components.p></>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = props.components || ({});
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`remark-frontmatter-additional > file with no title (expecting default first header to exist) 1`] = `
"/*@jsxRuntime automatic @jsxImportSource react*/
export const hello = \\"frontmatter\\", date = \\"2023-03-14\\", image = \\"https://cdn2.thecatapi.com/images/bdc.jpg\\";
export const excerpt = \\"hello world!\\";
export const slug = \\"/post-with-no-title\\";
export const readingTime = {
  \\"text\\": \\"1 min read\\",
  \\"minutes\\": 0.085,
  \\"time\\": 5100,
  \\"words\\": 17
};
export const title = \\"hello world!\\";
function _createMdxContent(props) {
  const _components = Object.assign({
    h1: \\"h1\\",
    blockquote: \\"blockquote\\",
    p: \\"p\\"
  }, props.components);
  return <><_components.h1>{\\"hello world!\\"}</_components.h1>{\\"\\\\n\\"}<_components.blockquote>{\\"\\\\n\\"}<_components.p>{\\"this is a simple post\\"}</_components.p>{\\"\\\\n\\"}</_components.blockquote></>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = props.components || ({});
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;
